core/src/Cabana_Sort.hpp

```diff
@@ -589,6 +589,195 @@ void permute( const BinningDataType& binning_data,
     Kokkos::fence();
 }
 
+//---------------------------------------------------------------------------//
+/*!
+  \brief Given binning data and range permute part of an AoSoA.
+
+  \tparam BinningDataType The binning data type.
+
+  \tparm AoSoA_t The AoSoA type.
+
+  \param binning_data The binning data.
+
+  \param aosoa The AoSoA to permute.
+
+  \param begin The beginning index of the AoSoA range to sort.
+
+  \param end The end index of the AoSoA range to sort.
+*/
+template<class BinningDataType, class AoSoA_t>
+void permute( const BinningDataType& binning_data,
```

Comment:
We can improve code reuse here. This should be the core implementation. Change the implementation of `permute` that permutes the entire AoSoA to just call this function with `aosoa.begin()` and `aosoa.end()`. That way we only have a single implementation. You can drop this call right in at line 566 and delete the rest.

Reply:
The original idea for this was to bin all particles with the `LinkedCellList`, permute only local or ghost, then reuse the binning with the `VerletList`. Since then we've decided against that, but that part was already done

Reply:
Removed